tsinfer/inference.py

```diff
@@ -153,6 +153,11 @@ def infer(
     :param int num_threads: The number of worker threads to use in parallelised
         sections of the algorithm. If <= 0, do not spawn any threads and
         use simpler sequential algorithms (default).
+    :param bool path_compression: Whether to merge edges that share identical
+        paths (essentially taking advantage of shared recombination breakpoints).
+    :param bool simplify: Whether to simply the resulting tree sequence using
+        :meth:`tskit.TreeSequence.simplify`, which ensures that it is in canonical form
```

Comment:
I guess the canonical form bit here is going to be misleading if we have ``keep_unary=True``. Probably worth making that change here if we're documenting this API?

Reply:
Yes, do you want me to change the simplify calling code, and therefore the docs too?

Reply:
Yes, this is probably a good thing to do. There'll be some breakage in the tests probably, though.

Reply:
Yep. I'm in the middle of breaking tests anyway. What's a few more to add to the mix...

Reply:
@jeromekelleher: how do we specify that we want to `keep_unary` in a call to `infer`. Should we have e.g. `simplify = None | (True) | (False)`, where `None` means `keep_unary` is passed to simplify? That seems a little obscure to me. Or should we have an option like `simplify_keep_unary = True | (False)`, and if people really want to do a full simplify, they can simply call it on the resulting TS?

Reply:
I'd say we either call simplify with ``keep_unary`` or we don't call simplify at all. If  the user wants the fully simplifed ts, then they can do that themselves easily enough.

Reply:
You mean we keep the parameter labelled `simplify` but just state in the docs that this means do so with `keep_unary=True`? Is there a problem with keeping that consistent with other usage?

Reply:
Exactly, yeah. We're still calling simplify, just with non-default parameters.